tests/cache/tests.py

```diff
@@ -602,12 +601,8 @@ def func(key, *args):
         cache.key_func = func
 
         try:
-            with warnings.catch_warnings(record=True) as w:
-                warnings.simplefilter("always")
+            with self.assertWarnsRegex(CacheKeyWarning, expected_warning):
```

Comment:
A better approach might be to pass `'^%s$' % re.escape(expected_warning))` to `assertWarnsRegex` instead of manually escaping.